uamqp/_connection.py

```diff
@@ -543,22 +565,34 @@ def listen(self, wait=False, batch=1, **kwargs):
          timeout (by default 0.1 seconds).
         :type wait: int or float or bool
         :param int batch: The number of frames to attempt to read and process before returning. The default value
-         is 1, i.e. process frames one-at-a-time. A higer value should only be used when a receiver is established
+         is 1, i.e. process frames one-at-a-time. A higher value should only be used when a receiver is established
          and is processing incoming Transfer frames.
         :rtype: None
         """
-        if self.state == ConnectionState.END:
-            raise ValueError("Connection closed.")
-        for _ in range(batch):
-            new_frame = self._read_frame(wait=wait, **kwargs)
-            if not new_frame:
+        try:
+            raise self._error
+        except TypeError:
+            pass
+        try:
+
+            if self.state == ConnectionState.END:
                 raise ValueError("Connection closed.")
-            if self._process_incoming_frame(*new_frame):
-                break
-        if self.state not in _CLOSING_STATES:
-            now = time.time()
-            if self._get_local_timeout(now) or self._get_remote_timeout(now):
-                self.close(error=None, wait=False)
+            for _ in range(batch):
+                new_frame = self._read_frame(wait=wait, **kwargs)
+                if not new_frame:
+                    raise ValueError("Connection closed.")
```

Comment:
Updated to be a `ConnectionClose` error.

Reply:
actually, I think here we should not raise a connection error just because we don't receive a frame -- chances are that a receiver keeps listening for frames while the service doesn't have any messages yet and besides we're in non-blocking mode, no frame is received in this iteration doesn't mean the connection is closed.